src/deepEqual.js

```diff
@@ -23,8 +23,45 @@
  *
  */
 
+function isEqual(value1, value2) {
+  return typeof value1 === typeof value2 && value1 === value2;
+}
+
+function isObject(elem) {
+  let result;
+  if (elem === null) {
+    result = false;
+  } else if (typeof elem === 'object') {
+    result = true;
+  } else {
+    result = false;
+  }
+  return result;
+}
+
+function isKeysLengthEqual(obj1, obj2) {
+  return Object.keys(obj1).length === Object.keys(obj2).length;
+}
+
 function deepEqual(a, b) {
-  // write code here
+  if (isObject(a) && isObject(b)) {
+    if (isKeysLengthEqual(a, b)) {
+      for (let key in a) {
+        let keyOfA = a[key];
```

Comment:
this is not really key, this is value for the key. So name should be valueOfA and valueOfB.